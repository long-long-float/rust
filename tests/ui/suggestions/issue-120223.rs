use std::{future::Future};

pub fn foo<T>(
    executor: impl FnOnce(T) -> dyn Future<Output = ()>,
) -> Box<dyn FnOnce(T) -> dyn Future<Output = ()>> {
    Box::new(executor) //~ ERROR the parameter type
}

fn main() {}
