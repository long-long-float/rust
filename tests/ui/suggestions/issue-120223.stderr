error[E0310]: the parameter type `impl FnOnce(T) -> dyn Future<Output = ()>` may not live long enough
  --> $DIR/issue-120223.rs:6:5
   |
LL |     Box::new(executor)
   |     ^^^^^^^^^^^^^^^^^^
   |     |
   |     the parameter type `impl FnOnce(T) -> dyn Future<Output = ()>` must be valid for the static lifetime...
   |     ...so that the type `impl FnOnce(T) -> dyn Future<Output = ()>` will meet its required lifetime bounds
   |
help: consider adding an explicit lifetime bound
   |
LL |     executor: impl FnOnce(T) -> (dyn Future<Output = ()>) + 'static,
   |                                 +                       +++++++++++

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0310`.
